core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -412,12 +412,6 @@ abstract class AbstractFetcherThread(name: String,
                        "expected to persist.")
                   partitionsWithError += topicPartition
 
-                case Errors.UNKNOWN_TOPIC_ID =>
```

Comment:
With the top level error change, we are no longer marking the partitions as `partitionsWithError`. I'm wondering if this is something we still want to do somehow to allow the broker's metadata time to update.